lib/optimizely/event_dispatcher.rb

```diff
@@ -28,26 +30,46 @@ class EventDispatcher
     # @api constants
     REQUEST_TIMEOUT = 10
 
+    def initialize(logger: nil, error_handler: nil)
+      @logger = logger || NoOpLogger.new
+      @error_handler = error_handler || NoOpErrorHandler.new
+    end
+
     # Dispatch the event being represented by the Event object.
     #
     # @param event - Event object
     def dispatch_event(event)
       if event.http_verb == :get
-        begin
-          HTTParty.get(event.url, headers: event.headers, query: event.params, timeout: REQUEST_TIMEOUT)
-        rescue Timeout::Error => e
-          return e
-        end
+        response = HTTParty.get(event.url, headers: event.headers, query: event.params, timeout: REQUEST_TIMEOUT)
+
       elsif event.http_verb == :post
-        begin
-          HTTParty.post(event.url,
-                        body: event.params.to_json,
-                        headers: event.headers,
-                        timeout: REQUEST_TIMEOUT)
-        rescue Timeout::Error => e
-          return e
-        end
+        response = HTTParty.post(event.url,
+                                 body: event.params.to_json,
+                                 headers: event.headers,
+                                 timeout: REQUEST_TIMEOUT)
       end
+
+      error_msg = "Event failed to dispatch with response code: #{response.code}"
```

Comment:
Shouldn't it be in constants file?

Reply:
We don't follow this as a practice in the SDK. And I would prefer to keep this here as I don't expect it to be reused.